src/nfa/nfa.cc

```diff
@@ -593,15 +593,17 @@ void Mata::Nfa::revert(Nfa* result, const Nfa& aut)
 
     if (aut.get_num_of_states() > result->get_num_of_states()) { result->increase_size(aut.get_num_of_states()); }
 
-    result->initialstates = aut.finalstates;
-    result->finalstates = aut.initialstates;
-
-    for (size_t i = 0; i < aut.get_num_of_states(); ++i)
+    for (State sourceState = 0; sourceState < aut.transitionrelation.size(); ++sourceState)
```

Comment:
When we are at it, could we extract the call for `aut.transitionrelation.size()` into a constant? We already do it everywhere, so…
```suggestion
    const size_t num_of_states{ aut.transitionrelation.size() };
    for (State sourceState = 0; sourceState < num_of_states; ++sourceState)
```